src/jdk.incubator.code/share/classes/jdk/incubator/code/interpreter/QuotableLambdaOpInvocationHandler.java

```diff
@@ -0,0 +1,30 @@
+package jdk.incubator.code.interpreter;
+
+import jdk.incubator.code.Quoted;
+
+import java.lang.reflect.InvocationHandler;
+import java.lang.reflect.Method;
+import java.util.Objects;
+
+public class QuotableLambdaOpInvocationHandler implements InvocationHandler {
```

Comment:
Can you enclose this class in the `Interpreter`? Did you try if it works as anonymous inner class of the `newProxyInstance` call?